specs/Print.md

```diff
@@ -0,0 +1,464 @@
+  /// `PrintWithSettings` is called while a `PrintWithSettings` or `PrintToPdf` or `PrintToPdfStream` or `Print`
+  /// job is in progress, the completed handler is immediately invoked with an error.
+  ///
+  /// \snippet AppWindow.cpp PrintWithSettings
+  HRESULT PrintWithSettings(
+    [in] ICoreWebView2PrintSettings2* printSettings,
+    [in] ICoreWebView2PrintWithSettingsCompletedHandler* handler);
+
+  /// Opens the print dialog to print the current page. See `COREWEBVIEW2_PRINT_DIALOG_KIND`
+  /// for descriptions of print dialog kinds.
+  /// The default value is `COREWEBVIEW2_PRINT_DIALOG_KIND_BROWSER_PRINT_PREVIEW`.
+  ///
+  /// \snippet AppWindow::OpenPrintDialog
+  HRESULT Print([in] COREWEBVIEW2_PRINT_DIALOG_KIND printDialogKind);
+
+  /// Provides the Pdf data of current page asynchronously for the provided settings.
+  /// See `ICoreWebView2PrintSettings2` for description of settings. Passing
+  /// nullptr for `printSettings` results in default print settings used.
+  ///
+  /// The async `PrintToPdfStream` operation completes when it finishes
+  /// writing to the stream. At this time the `ICoreWebView2PrintToPdfStreamCompletedHandler`
+  /// is invoked. Only one `Printing` operation can be in progress at a time. If
+  /// `PrintToPdfStream` is called while a `PrintToPDFStream` or `PrintToPdf` or `PrintWithSettings` or `Print`
+  /// job is in progress, the completed handler is immediately invoked with an error.
+  ///
+  /// \snippet AppWindow.cpp PrintToPdfStream
+  HRESULT PrintToPdfStream([in] ICoreWebView2PrintSettings2* printSettings,
+                           [in] ICoreWebView2PrintToPdfStreamCompletedHandler* handler);
+}
+
+/// Receives the result of the `PrintWithSettings` method.
+/// `errorCode` returns S_OK if the PrintWithSettings operation succeeded.
+[uuid(4B2FE043-1A1A-4924-803D-C05E5FDCA3B0), object, pointer_default(unique)]
+interface ICoreWebView2PrintWithSettingsCompletedHandler : IUnknown {
+  /// Provides the result of the corresponding asynchronous method.
+  HRESULT Invoke([in] HRESULT errorCode);
+}
+
+/// Receives the result of the `PrintToPdfStream` method.
+/// `errorCode` returns S_OK if the PrintToPdfStream operation succeeded.
+/// The printable pdf data is returned in the `pdfStream` object.
+[uuid(78A9FC37-249B-4E7E-A2E6-20CAAA14AE7B), object, pointer_default(unique)]
+interface ICoreWebView2PrintToPdfStreamCompletedHandler : IUnknown {
+  /// Provides the result of the corresponding asynchronous method.
+  HRESULT Invoke([in] HRESULT errorCode, [in] IStream* pdfStream);
+}
+
+/// Settings used by the `PrintWithSettings` method.
+[uuid(DA3AD25C-D121-44DC-83DA-BF5E50DBF0DF), object, pointer_default(unique)]
+interface ICoreWebView2PrintSettings2 : ICoreWebView2PrintSettings {
+  /// Page range to print. Defaults to empty string, which means print all pages.
+  ///
+  /// Print ranges should be separated by commas.
```

Comment:
I feel like we're missing some context here. Also the term 'Print ranges' doesn't show up again in this doc. Maybe this is supposed to be Page ranges? Also, something about whitespace (is the following true?)

```suggestion
  /// The PageRanges property is a list of page ranges specifying one or more pages that should be printed separated by commas. Any whitespace between page ranges is ignored.
```